src/LEGO.AsyncAPI/Models/AsyncApiReference.cs

```diff
@@ -67,6 +95,21 @@ public void SerializeV2(IAsyncApiWriter writer)
             writer.WriteEndObject();
         }
 
+        private string GetExternalReferenceV2()
```

Comment:
Not sure I understand what intention is with this method. If, its external why does it start pointing an `id` in `components` on the document?

Reply:
There is a case, where the reference is a full asyncapidocument, and we are pointing at a 'real' reference there.
So it sets up the Reference to match what the internal references look like, except it has the 'externalResource' set as the document path.  made a test case for it